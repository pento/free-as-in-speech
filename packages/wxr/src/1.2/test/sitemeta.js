/**
 * External dependencies
 */
const FDBFactory = require( 'fake-indexeddb/lib/FDBFactory' );

/**
 * Internal dependencies
 */
const { WXRDriver } = require( '../index.js' );

describe( 'Site Meta', () => {
	beforeEach( () => {
		window.indexedDB = new FDBFactory();
	} );

	test( 'All fields are written as expected', async () => {
		const wxr = new WXRDriver();
		await wxr.connect();

		wxr.setSiteMeta( {
			title: 'A Title',
			link: 'https://wordpress.org/',
			description: 'A description',
			language: 'art-x-emoji',
			wxrVersion: '1.2.3.4.5', // Should not be writable, test with a fake value.
			siteUrl: 'https://make.wordpress.org/',
			blogUrl: 'https://wordpress.org/news/',
		} );

		const xml = await wxr.export();

		const cleanedXml = xml.replace(
			/<pubDate>.*?<\/pubDate>/,
			'<pubDate></pubDate>'
		);

		expect( cleanedXml ).toMatchSnapshot();
	} );

	test( 'siteUrl and BlogUrl copy from link', async () => {
		const wxr = new WXRDriver();
		await wxr.connect();

		wxr.setSiteMeta( {
			link: 'https://wordpress.org/',
		} );

		const xml = await wxr.export();

		const cleanedXml = xml.replace(
			/<pubDate>.*?<\/pubDate>/,
			'<pubDate></pubDate>'
		);

		expect( cleanedXml ).toMatchSnapshot();
	} );
} );
