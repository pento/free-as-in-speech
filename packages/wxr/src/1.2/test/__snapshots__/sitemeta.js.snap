// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site Meta All fields are written as expected 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>

<!-- This is a WordPress eXtended RSS file generated by the \\"Free (as in Speech)\\" browser extension, as an export of your site. -->
<!-- It contains information about your site's posts, pages, comments, categories, and other content. -->
<!-- You may use this file to transfer that content from one site to another. -->
<!-- This file is not intended to serve as a complete backup of your site. -->

<!-- To import this information into a WordPress site follow these steps: -->
<!-- 1. Log in to that site as an administrator. -->
<!-- 2. Go to Tools: Import in the WordPress admin panel. -->
<!-- 3. Install the \\"WordPress\\" importer from the list. -->
<!-- 4. Activate & Run Importer. -->
<!-- 5. Upload this file using the form provided on that page. -->
<!-- 6. You will first be asked to map the authors in this export file to users -->
<!--    on the site. For each author, you may choose to map to an -->
<!--    existing user on the site or to create a new user. -->
<!-- 7. WordPress will then import each of the posts, pages, comments, categories, etc. -->
<!--    contained in this file into your site. -->

<rss version=\\"2.0\\"
	xmlns:excerpt=\\"http://wordpress.org/export/1.2/excerpt/\\"
	xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\"
	xmlns:wfw=\\"http://wellformedweb.org/CommentAPI/\\"
	xmlns:dc=\\"http://purl.org/dc/elements/1.1/\\"
	xmlns:wp=\\"http://wordpress.org/export/1.2/\\"
>
<channel>
	<title><![CDATA[A Title]]></title>
	<link><![CDATA[https://wordpress.org/]]></link>
	<description><![CDATA[A description]]></description>
	<pubDate></pubDate>
	<language><![CDATA[art-x-emoji]]></language>
	<wp:wxr_version>1.2</wp:wxr_version>
	<wp:base_site_url><![CDATA[https://make.wordpress.org/]]></wp:base_site_url>
	<wp:base_blog_url><![CDATA[https://wordpress.org/news/]]></wp:base_blog_url>
</channel>
</rss>"
`;

exports[`Site Meta siteUrl and BlogUrl copy from link 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>

<!-- This is a WordPress eXtended RSS file generated by the \\"Free (as in Speech)\\" browser extension, as an export of your site. -->
<!-- It contains information about your site's posts, pages, comments, categories, and other content. -->
<!-- You may use this file to transfer that content from one site to another. -->
<!-- This file is not intended to serve as a complete backup of your site. -->

<!-- To import this information into a WordPress site follow these steps: -->
<!-- 1. Log in to that site as an administrator. -->
<!-- 2. Go to Tools: Import in the WordPress admin panel. -->
<!-- 3. Install the \\"WordPress\\" importer from the list. -->
<!-- 4. Activate & Run Importer. -->
<!-- 5. Upload this file using the form provided on that page. -->
<!-- 6. You will first be asked to map the authors in this export file to users -->
<!--    on the site. For each author, you may choose to map to an -->
<!--    existing user on the site or to create a new user. -->
<!-- 7. WordPress will then import each of the posts, pages, comments, categories, etc. -->
<!--    contained in this file into your site. -->

<rss version=\\"2.0\\"
	xmlns:excerpt=\\"http://wordpress.org/export/1.2/excerpt/\\"
	xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\"
	xmlns:wfw=\\"http://wellformedweb.org/CommentAPI/\\"
	xmlns:dc=\\"http://purl.org/dc/elements/1.1/\\"
	xmlns:wp=\\"http://wordpress.org/export/1.2/\\"
>
<channel>
	<link><![CDATA[https://wordpress.org/]]></link>
	<pubDate></pubDate>
	<wp:wxr_version>1.2</wp:wxr_version>
	<wp:base_site_url><![CDATA[https://wordpress.org/]]></wp:base_site_url>
	<wp:base_blog_url><![CDATA[https://wordpress.org/]]></wp:base_blog_url>
</channel>
</rss>"
`;
